VERSIONING.md

```diff
@@ -1,13 +1,42 @@
-# Versioning
+# Versioning and stability
 
-This document describes the versioning policy for this repository. This policy
-is designed so that the following goal can be achieved:
+The OpenTelemetry Collector SIG produces several artifacts for [a variety of audiences](CONTRIBUTING.md#target-audiences). This document describes the versioning and support policy for these artifacts. These policies are designed so that the following goal can be achieved:
 
-**Users are provided a codebase of value that is stable and secure.**
+**Users are provided software artifacts of value that are stable and secure.**
 
-## Public API expectations
+The policies are divided depending on the artifact's target audience. While an artifact is supported, [critical bugs](docs/release.md#bugfix-release-criteria) and security vulnerabilities MUST be addressed. The main criteria for the length of support for an artifact is how easy it is for an artifact's target audience to adapt to disruptive changes.
 
-The following public API expectations apply to all modules in opentelemetry-collector and opentelemetry-collector-contrib.
+These policies reflect the current consensus of the OpenTelemetry Collector SIG. They are subject to change as the project evolves.
+
+## Software artifacts for end users
+
+Software artifacts intended for [end users](CONTRIBUTING.md#end-users) of the OpenTelemetry Collector include 
+- Binary distributions of the OpenTelemetry Collector.
+- Go modules that expose Collector components, such as receivers, processors, connectors, extensions and exporters. 
```

Comment:
This mentions Go modules for components. What about all other Go modules published by the project, like e.g. https://pkg.go.dev/go.opentelemetry.io/collector/otelcol or https://pkg.go.dev/go.opentelemetry.io/collector/config/confighttp? Do they get a different treatment than the Go modules for components?

Reply:
Yes, they have a smaller support window https://github.com/open-telemetry/opentelemetry-collector/blob/8205489835974f8599e3553e03819ad8a0b6a5db/VERSIONING.md?plain=1#L173 

The rationale for this is that they are equipped to deal with more disruptive changes https://github.com/open-telemetry/opentelemetry-collector/blob/012cf862283a1af411c5b1ace844a6c1d7a82766/CONTRIBUTING.md?plain=1#L45-L46